src/main/java/com/playground/playground/FeaturesHiddenLayersController.java

```diff
@@ -0,0 +1,196 @@
+    @FXML
+    private Button removeLayer;
+    @FXML
+    private Button x1pow2button;
+    @FXML
+    private Button x2pow2button;
+    @FXML
+    private Button x1x2button;
+    @FXML
+    private Button sinx1button;
+    @FXML
+    private Button sinx2button;
+    @FXML
+    private Button add1;
+    @FXML
+    private Button remove1;
+    @FXML
+    private Button add2;
+    @FXML
+    private Button remove2;
+    @FXML
+    private Button add3;
+    @FXML
+    private Button remove3;
+    @FXML
+    private Button add4;
+    @FXML
+    private Button remove4;
+    @FXML
+    private Button add5;
+    @FXML
+    private Button remove5;
+    @FXML
+    private Button add6;
+    @FXML
+    private Button remove6;
+    @FXML
+    public GridPane neuralConnections;
+    private Button[] addButtons;
+    private Button[] removeButtons;
+    private Button[][] aButtons;
+    private int[] aButtonCounts;
+    private int i = 0;
+
+    @FXML
+    private Label neurons1, neurons2, neurons3, neurons4, neurons5, neurons6;
+
+    private Label[] neuronLabels;
+
+    /**
+     * Initializer for FeaturesHiddenLayersController.java
+     * @param location
+     * The location used to resolve relative paths for the root object, or
+     * {@code null} if the location is not known.
+     *
+     * @param resources
+     * The resources used to localize the root object, or {@code null} if
+     * the root object was not localized.
+     */
+    public void initialize(URL location, ResourceBundle resources) {
+        setButtonWithImage(x1pow2button, "playground-images/x1pow2button.jpg");
+        setButtonWithImage(x2pow2button, "playground-images/x2pow2button.jpg");
+        setButtonWithImage(x1x2button, "playground-images/x1x2button.jpg");
+        setButtonWithImage(sinx1button, "playground-images/sinx1button.jpg");
+        setButtonWithImage(sinx2button, "playground-images/sinx2button.jpg");
+
+        addButtons = new Button[]{add1, add2, add3, add4, add5, add6};
+        removeButtons = new Button[]{remove1, remove2, remove3, remove4, remove5, remove6};
+        aButtons = new Button[6][8];
+        aButtonCounts = new int[6];
+        neuronLabels = new Label[]{neurons1, neurons2, neurons3, neurons4, neurons5, neurons6};
+    }
+
+    private void setButtonWithImage(Button button, String imagePath) {
+        ImageView imageView = new ImageView(getClass().getResource(imagePath).toExternalForm());
+        imageView.setFitWidth(35);
+        imageView.setFitHeight(35);
+        button.setGraphic(imageView);
+        button.getStyleClass().add("image-button");
+    }
+
+    /**
+     * Adds a layer to the hidden layer graph and increases the value of the layer counter by 1.
+     * @param event is triggered by user interaction with the add layer button.
+     */
+    @FXML
+    private void onAddLayerClicked(ActionEvent event) {
+        int currentCount = Integer.parseInt(numHiddenLayers.getText());
+        if (currentCount < 6 && i < 6) {
```

Comment:
I think it would be worthwhile to display a message saying that the app only supports up to six layers at the time, so that the users don't feel the app is unresponsive.

Reply:
+1 on this